### PR TITLE
Part 4 of unit test coverage + tiny readme update to add developer agreement link

### PR DESCRIPTION
**Pull request by priyankalakhe**
# Description of changes

- More unit test coverage (followup to https://github.com/smartsheet/smartsheet-java-sdk/pull/69) 
- Copy over changes from https://github.com/smartsheet/smartsheet-java-sdk/pull/17/files

### CODE REVIEW

File: src/test/java/com/smartsheet/api/internal/SightResourcesImplTest.java [hunk 1]
```diff
@@ -36,9 +50,145 @@ public void before() {
...
+        server.setResponseBody(new File("src/test/resources/updateSight.json"));
+        assertThatCode(() -> sightResourcesImpl.deleteSight(12345L)).doesNotThrowAnyException();
+    }
+
+    @Test
+    void testSetPublishStatus() throws IOException, SmartsheetException {
+        server.setResponseBody(new File("src/test/resources/setSightPublishStatus.json"));
+        SightPublish sightPublish = sightResourcesImpl.setPublishStatus(1234L, new SightPublish());
+        assertThat(sightPublish.getReadOnlyFullAccessibleBy()).isEqualTo(Boolean.FALSE.toString());
+        assertThat(sightPublish.getReadOnlyFullAccessibleBy()).isEqualTo(Boolean.FALSE.toString());
```
Comment (zromano) [inline src/test/java/com/smartsheet/api/internal/SightResourcesImplTest.java]: nit: same assertion twice here

File: src/test/java/com/smartsheet/api/internal/SightResourcesImplTest.java [hunk 2]
```diff
@@ -36,9 +50,145 @@ public void before() {
...
+        assertThat(sightPublish.getReadOnlyFullUrl()).isNotBlank();
+    }
+
+    @Test
+    void testGetPublishStatus() throws IOException, SmartsheetException {
+        server.setResponseBody(new File("src/test/resources/getSightPublishStatus.json"));
+        SightPublish sightPublish = sightResourcesImpl.getPublishStatus(1234L);
+        assertThat(sightPublish.getReadOnlyFullAccessibleBy()).isEqualTo(Boolean.FALSE.toString());
```
Comment (zromano) [inline src/test/java/com/smartsheet/api/internal/SightResourcesImplTest.java:164]: nit: same assertion twice here

File: src/test/java/com/smartsheet/api/internal/UserResourcesImplTest.java [hunk 3]
```diff
@@ -144,6 +151,29 @@ void testGetCurrentUser() throws SmartsheetException, IOException {
...
+        assertThat(user.getLastName()).isEqualTo("Doe");
+        assertThat(user.getLocale()).isEqualTo("en_US");
+        assertThat(user.getTimeZone()).isEqualTo("US/Pacific");
+
+        Account account = user.getAccount();
+        assertThat(account.getName()).isEqualTo("Smartsheet");
+        assertThat(account.getId().longValue()).isEqualTo(111111111111L);
+
+        // has groups
+        assertThat(user.getGroups().size()).isEqualTo(1);
```
Comment (zromano) [inline src/test/java/com/smartsheet/api/internal/UserResourcesImplTest.java]: nit: you can do  `assertThat(user.getGroups()).hasSize(1);` here (and other places we are asserting size)

File: src/test/java/com/smartsheet/api/internal/UserResourcesImplTest.java [hunk 4]
```diff
@@ -177,5 +207,124 @@ void testListOrgSheets() throws SmartsheetException, IOException {
...
+        server.setResponseBody(new File("src/test/resources/listAlternateEmails.json"));
+
+        PaginationParameters pagination = new PaginationParameters();
+        pagination.setIncludeAll(true);
+        pagination.setPageSize(1);
+        pagination.setPage(1);
+
+        PagedResult<AlternateEmail> alternateEmailPagedResult = userResources.listAlternateEmails(1234L, pagination);
+        assertThat(alternateEmailPagedResult.getData()).isNotNull();
+        assertThat(alternateEmailPagedResult.getData()).isNotEmpty();
```
Comment (zromano) [inline src/test/java/com/smartsheet/api/internal/UserResourcesImplTest.java]: nit: these assertions can be chained:
```java
assertThat(alternateEmailPagedResult.getData()).isNotNull().isNotEmpty();
```

**Review by zromano — DISMISSED** [review]
Love it. Thanks for doing this

File: src/test/java/com/smartsheet/api/internal/UserResourcesImplTest.java (hunk 4 above)
Comment (priyankalakhe) [reply src/test/java/com/smartsheet/api/internal/UserResourcesImplTest.java]: Fixed as part of https://github.com/smartsheet/smartsheet-java-sdk/pull/74/commits/34b4b8734054fed5a7fc11ecc85d8dbf93e600d2
Resolving!

File: src/test/java/com/smartsheet/api/internal/UserResourcesImplTest.java (hunk 3 above)
Comment (priyankalakhe) [reply src/test/java/com/smartsheet/api/internal/UserResourcesImplTest.java]: Fixed as part of https://github.com/smartsheet/smartsheet-java-sdk/pull/74/commits/34b4b8734054fed5a7fc11ecc85d8dbf93e600d2
Resolving!

File: src/test/java/com/smartsheet/api/internal/SightResourcesImplTest.java (hunk 1 above)
Comment (priyankalakhe) [reply src/test/java/com/smartsheet/api/internal/SightResourcesImplTest.java]: Fixed as part of https://github.com/smartsheet/smartsheet-java-sdk/pull/74/commits/34b4b8734054fed5a7fc11ecc85d8dbf93e600d2
Resolving!

File: src/test/java/com/smartsheet/api/internal/SightResourcesImplTest.java (hunk 2 above)
Comment (priyankalakhe) [reply src/test/java/com/smartsheet/api/internal/SightResourcesImplTest.java:164]: Fixed as part of https://github.com/smartsheet/smartsheet-java-sdk/pull/74/commits/34b4b8734054fed5a7fc11ecc85d8dbf93e600d2
Resolving!

**Review by zromano — APPROVED** [review]

**Pull request merged**